app/services/classification/base.rb

```diff
@@ -27,10 +27,8 @@ def class_parameters
     end
 
     def possible_value_orders
-      class_parameters.select do |key|
-        # take only keys with actual values
-        key.lookup_values_count > 0 # we use counter cache, so its safe to make that query
-      end.map(&:path_elements).flatten(1).uniq
+      # the inner join makes sure we take only keys with actual values
+      class_parameters.joins(:lookup_values).flat_map(&:path_elements).uniq
     end
```

Comment:
This is a helper function used for calculating which matchers apply to a certain host, it is exercised indirectly by the classification test.